core/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoHashGridParser.java

```diff
@@ -77,14 +76,18 @@ public AggregatorFactory parse(String aggregationName, XContentParser parser, Se
                 currentFieldName = parser.currentName();
             } else if (vsParser.token(currentFieldName, token, parser)) {
                 continue;
-            } else if (token == XContentParser.Token.VALUE_NUMBER) {
-                if ("precision".equals(currentFieldName)) {
-                    precision = parser.intValue();
-                } else if ("size".equals(currentFieldName)) {
+            } else if (token == XContentParser.Token.VALUE_NUMBER ||
+                    token == XContentParser.Token.VALUE_STRING) { //Be lenient and also allow numbers enclosed in quotes
```

Comment:
We cannot use `#isValue()` here because it would also return `true` on other types such as booleans which we want to exclude.